### PR TITLE
Bugfix/tor 0.3.15

### PR DESCRIPTION
**Pull request by ProofOfKeags**

### CODE REVIEW

File: agent/src/Lib/Synchronizers.hs [hunk 1]
```diff
@@ -585,19 +589,35 @@ syncRestarterService = SyncOp "Install Restarter Service" check migrate True
...
-                        $> False
-                        )
-                `catch` \(e :: ProcessException) -> case e of
-                            ProcessException _ (ExitFailure 1) -> pure True
-                            _ -> throwIO e
+        check = run $ do
+            shell "apt-get clean"
+            shell "apt-get update"
+            mTorVersion <- (shell "dpkg -s tor" $| shell "grep '^Version'" $| shell "cut -d ' ' -f2" $| conduit await)
+            echo ("CURRENT TOR VERSION:" :: Text) (show mTorVersion :: Text)
```
Comment (ProofOfKeags) [inline agent/src/Lib/Synchronizers.hs:598]: delete

File: agent/src/Lib/Synchronizers.hs [hunk 2]
```diff
@@ -585,19 +589,35 @@ syncRestarterService = SyncOp "Install Restarter Service" check migrate True
...
+                    Just x  -> x
+            shell $ "apt-get install -y tor=" <> if "0.3.5.15-1" `elem` availVersions
+                then "0.3.5.15-1"
+                else (C8.unpack latest)
+        compareTorVersions :: ByteString -> ByteString -> Ordering
+        compareTorVersions a b =
+            let a' = (traverse (readMaybe @Int . decodeUtf8) . (split '.' <=< split '-') $ a)
+                b' = (traverse (readMaybe @Int . decodeUtf8) . (split '.' <=< split '-') $ b)
+            in  case liftA2 compare a' b' of
+                    Nothing -> panic "invalid tor version string"
```
Comment (ProofOfKeags) [inline agent/src/Lib/Synchronizers.hs:618]: forgive my sins.

File: agent/src/Lib/Synchronizers.hs [hunk 3]
```diff
@@ -585,19 +589,35 @@ syncRestarterService = SyncOp "Install Restarter Service" check migrate True
...
+                    Nothing -> panic "invalid output from dpkg, can't read tor version"
+                    Just x  -> x
+            pure $ compareTorVersions torVersion "0.3.5.15-1" == LT
         migrate = liftIO . run $ do
             shell "apt-get update"
-            shell "apt-get install -y tor=0.3.5.14-1"
+            availVersions <-
+                (shell "apt-cache madison tor" $| shell "cut -d '|' -f2" $| shell "xargs" $| conduit consume)
+            let latest = case lastMay $ sortBy compareTorVersions availVersions of
+                    Nothing -> panic "No available versions of tor"
```
Comment (ProofOfKeags) [inline agent/src/Lib/Synchronizers.hs:608]: this need not be a panic

**Review by kn0wmad — COMMENTED** [review]
lgtm

**Review by elvece — APPROVED** [review]

**Pull request merged**